player.gd

```diff
@@ -230,11 +247,13 @@ func _rollback_tick(delta, _tick, _is_fresh):
 				movement_state = MovementState.JUMPING
 
 		MovementState.CROUCH_IDLE:
+			if input.direction.x == 0 and is_on_floor():
+				velocity.x = move_toward(velocity.x, 0, ground_friction + (ground_friction * crouch_penalty))
 			if input.direction.x != 0:
 				movement_state = MovementState.CROUCH_WALK
 			if not is_on_floor():
 				movement_state = MovementState.JUMPING
-			if not input.down[1]:
+			if input.down[2]:
```

Comment:
Is this the fix to the infinite velocity glitch?

Reply:
yes, i don't remember my exact thought process but it was something about the not statement being explicit enough and you could still retain your velocity